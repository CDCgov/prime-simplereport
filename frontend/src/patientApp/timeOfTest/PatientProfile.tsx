import moment from "moment";
import { Navigate } from "react-router-dom";
import { useSelector } from "react-redux";

import { formatFullName } from "../../app/utils/user";
import { formatAddress, newLineSpan } from "../../app/utils/address";
import {
  RACE_VALUES,
  ETHNICITY_VALUES,
  GENDER_VALUES,
  TRIBAL_AFFILIATION_VALUES,
} from "../../app/constants";

interface Props {
  patient: any;
}

const PatientProfile = ({ patient }: Props) => {
  const plid = useSelector((state: any) => state.plid);
  if (!patient) {
    return (
      <Navigate
        to={{
          pathname: "/pxp",
          search: `?plid=${plid}`,
        }}
      />
    );
  }
  const fullName = formatFullName(patient);
  const race = RACE_VALUES.find((val) => val.value === patient.race)?.label;
  const ethnicity = ETHNICITY_VALUES.find(
    (val) => val.value === patient.ethnicity
  )?.label;
  const tribalAffiliation =
    patient.tribalAffiliation?.length &&
    TRIBAL_AFFILIATION_VALUES.find(
      (val) => val.value === patient.tribalAffiliation[0]
    )?.label;
  const gender = GENDER_VALUES.find((val) => val.value === patient.gender)
    ?.label;

  const address = formatAddress({
    street: patient.street,
    streetTwo: patient.streetTwo,
    city: patient.city,
    state: patient.state,
    zipCode: patient.zipCode,
  });
  const notProvided = "Not provided";

  return (
    <div className="prime-formgroup usa-prose">
      <h2 className="prime-formgroup-heading font-heading-lg">
        General information
      </h2>
      <h3 className="font-heading-sm">Name</h3>
      <p>{fullName}</p>
      <h3 className="font-heading-sm">Preferred language</h3>
      <p>{patient.preferredLanguage || notProvided}</p>
      <h3 className="font-heading-sm">Date of birth</h3>
      <p>
        {patient.birthDate
          ? moment(patient.birthDate).format("MM/DD/yyyy")
          : notProvided}
      </p>
      <h3 className="font-heading-sm">Phone number</h3>
      <p>{patient.telephone || notProvided}</p>
      {/* <h3 className="font-heading-sm">Phone type</h3>
      <p></p> */}
      <h3 className="font-heading-sm">Address</h3>
      <p>{address ? newLineSpan({ text: address }) : notProvided}</p>
      <h3 className="font-heading-sm">Email address</h3>
      <p id="patient-email">{patient.email || notProvided}</p>
      <h2 className="prime-formgroup-heading font-heading-lg">Demographics</h2>
      <h3 className="font-heading-sm">Race</h3>
      <p>{race || notProvided}</p>
      <h3 className="font-heading-sm">Ethnicity</h3>
      <p>{ethnicity || notProvided}</p>
      <h3 className="font-heading-sm">Tribal affiliation</h3>
      <p>{tribalAffiliation || notProvided}</p>
      <h3 className="font-heading-sm">Sex assigned at birth</h3>
      <p>{gender || notProvided}</p>
      <h2 className="prime-formgroup-heading font-heading-lg">Other</h2>
      <h3 className="font-heading-sm">
        Are you a resident in a congregate living setting?
      </h3>
      <p>
        {patient.residentCongregateSetting === true
          ? "Yes"
          : patient.residentCongregateSetting === false
          ? "No"
          : notProvided}
      </p>
      <h3 className="font-heading-sm">Are you a health care worker?</h3>
      <p>
        {patient.employedInHealthcare === true
          ? "Yes"
          : patient.employedInHealthcare === false
          ? "No"
          : notProvided}
      </p>
    </div>
  );
};

export default PatientProfile;
