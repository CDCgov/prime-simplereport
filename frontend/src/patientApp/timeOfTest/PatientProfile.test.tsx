import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render, screen } from "@testing-library/react";

import PatientProfile from "./PatientProfile";

jest.mock("react-router-dom", () => {
  const original = jest.requireActual("react-router-dom");
  return {
    ...original,
    Navigate: () => <p>Redirected</p>,
  };
});

const mockStore = configureStore([]);

describe("PatientProfile", () => {
  it("renders", () => {
    const patient = {
      firstName: "Jane",
      lastName: "Doe",
      street: "24 Dreary Ln",
    };
    const store = mockStore({
      plid: "foo",
    });
    render(
      <Provider store={store}>
        <PatientProfile patient={patient} />
      </Provider>
    );
    expect(
      screen.getByText("General information", { exact: false })
    ).toBeInTheDocument();
  });
  it("should redirect to '/' if no patient", () => {
    const store = mockStore({
      plid: "foo",
    });
    render(
      <Provider store={store}>
        <PatientProfile patient={null} />
      </Provider>
    );
    // eslint-disable-next-line no-restricted-globals
    expect(
      screen.getByText("Redirected", { exact: false })
    ).toBeInTheDocument();
  });
});
