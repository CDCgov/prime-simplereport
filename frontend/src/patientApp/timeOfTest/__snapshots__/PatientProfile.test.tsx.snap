// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientProfile snapshot 1`] = `
<div
  className="prime-formgroup usa-prose"
>
  <h2
    className="prime-formgroup-heading font-heading-lg"
  >
    General information
  </h2>
  <h3
    className="font-heading-sm"
  >
    Name
  </h3>
  <p>
    Jane Doe
  </p>
  <h3
    className="font-heading-sm"
  >
    Preferred language
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Date of birth
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Phone number
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Address
  </h3>
  <p>
    <span
      className="display-block"
    >
      24 Dreary Ln
    </span>
    <span
      className="display-block"
    >
      undefinedundefined
    </span>
  </p>
  <h3
    className="font-heading-sm"
  >
    Email address
  </h3>
  <p
    id="patient-email"
  >
    Not provided
  </p>
  <h2
    className="prime-formgroup-heading font-heading-lg"
  >
    Demographics
  </h2>
  <h3
    className="font-heading-sm"
  >
    Race
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Ethnicity
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Tribal affiliation
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Sex assigned at birth
  </h3>
  <p>
    Not provided
  </p>
  <h2
    className="prime-formgroup-heading font-heading-lg"
  >
    Other
  </h2>
  <h3
    className="font-heading-sm"
  >
    Are you a resident in a congregate living setting?
  </h3>
  <p>
    Not provided
  </p>
  <h3
    className="font-heading-sm"
  >
    Are you a health care worker?
  </h3>
  <p>
    Not provided
  </p>
</div>
`;
